srml/sudo/src/lib.rs

```diff
@@ -14,8 +14,106 @@
+//! 
+//! The sudo module allows for a single account to be set as the sudo key of your blockchain. The sudo key can execute dispatchable functions which require a `Root` call or assign a new account to replace them as the sudo key.
+//! 
+//! You can start using the Sudo module by implementing the sudo [`Trait`].
+//! 
+//! Supported dispatchables functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable
+//! 
+//! Only the sudo key can call the dispatchable functions from the Sudo module.
+//! 
+//! * `sudo` - Make a `Root` call to a dispatchable function.
+//! * `set_key` - Assign a new account to be the key of the superuser
+//! 
+//! Please refer to the [`Call`] enum and its associated variants for a detailed list of dispatchable functions.
+//!
+//! ## Usage
+//! 
+//! ### Prerequisites
+//! 
+//! To use the sudo module in your runtime, you must implement the following trait in your runtime:
+//! 
+//! ```ignore
+//! impl sudo::Trait for Runtime {
+//! 	/// The uniquitous event type.
+//! 	type Event = Event;
+//! 	type Proposal = Call;
+//! }
+//! ```
+//! 
+//! You can then import the sudo module in your `construct_runtime!` macro with:
+//! 
+//! ```ignore
+//! Sudo: sudo,
+//! ```
+//! 
+//! ### Simple Code Snippet
+//! 
+//! The Sudo module itself is not intended to be used within other modules. However, you can build other dispatchable functions to take advantage of this sudo module by requiring the function only accept calls from `Root`.
+//! 
+//! For example:
+//! 
+//! ```ignore
+//! use support::{decl_module, dispatch::Result};
+//! use system::ensure_root;
+//! 
+//! pub trait Trait: system::Trait {}
+//! 
+//! decl_module! {
+//!     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+//!         pub fn privileged_function(origin) -> Result {
+//!             ensure_root(origin)?;
+//!             
+//!             // do something...
+//! 
+//!             Ok(())
+//!         }
+//!     }
+//! }
+//! ```
+//! 
+//! This `privileged_function()` can not directly be executed via an extrinsic, but can be called via `sudo()` function.
+//! 
+//! 
+//! ### Example from SRML
+//! 
+//! The consensus module exposes a `set_code()` function which requires a `Root` call and allows you to set the on-chain Wasm runtime code:
+//! 
+//! ```ignore
+//! /// Set the new code.
+//! pub fn set_code(new: Vec<u8>) {
+//!     storage::unhashed::put_raw(well_known_keys::CODE, &new);
+//! }
+//! ```
+//! 
+//! Note that this function does not include the `origin` parameter, so the `decl_module` macro adds `origin` and the `ensure_root` check automatically.
+//! 
+//! ## Genesis Config
+//! 
+//! To use the sudo module, you need to include an initial superuser account to be set as the sudo `key`.
+//! 
+//! ```ignore
+//! GenesisConfig {
+//!     sudo: Some(SudoConfig {
+//!         key: AccountId,
+//!     })
+//! }
+//! ```
+//! 
+//! ## Related Modules
+//! 
+//! * Consensus
```

Comment:
link to docs

Reply:
I dunno how :(

Reply:
Not as elegant, but can just link to the http site? 
https://crates.parity.io/srml_consensus/index.html

Reply:
You can use HTML `<a href="https://crates.parity.io/srml_consensus/index.html" target="_blank">SRML Consensus</a>`

I did it here a lot https://github.com/paritytech/substrate/pull/1947/files

Reply:
I just hate the idea of using hard links. (btw you should be able to just use markdown right?)

Reply:
Yes, regular markdown works. I don't think hard links should prevent these from going through. It is still a link to rustdocs that should be kept up to date with the modules to which they are linking. If you want to link to the user's generated rustdoc (and not what's hosted on crates.parity.io), then you could use relative links (../../srml_consensus/index.html).